Assets/MixedRealityToolkit/_Core/Interfaces/IMixedRealityController.cs

```diff
@@ -1,48 +1,49 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem;
 using UnityEngine;
 
 namespace Microsoft.MixedReality.Toolkit.Internal.Definitions
 {
     /// <summary>
     /// The Controller definition defines the Controller as defined by the SDK / Unity.
     /// </summary>
-    public struct Controller
+    public interface IMixedRealityController<T> : IMixedRealityInputSource
     {
         /// <summary>
-        /// The ID assigned to the Controller
+        /// The designated hand that the controller is managing, as defined by the SDK / Unity.
         /// </summary>
-        public string Id { get; set; }
+        Handedness Handedness { get; }
 
         /// <summary>
-        /// The designated hand that the controller is managing, as defined by the SDK / Unity.
+        /// Outputs the current position of the controller, as defined by the SDK / Unity.
         /// </summary>
-        public Handedness Handedness { get; set; }
+        T Controller { get; }
 
         /// <summary>
-        /// Outputs the current position of the controller, as defined by the SDK / Unity.
+        /// Outputs the current Pointer position of the controller, as defined by the SDK / Unity.
         /// </summary>
-        public Vector3 Position { get; set; }
+        T Pointer { get; }
 
         /// <summary>
-        /// Outputs the current rotation of the controller, as defined by the SDK / Unity.
+        /// Outputs the current Grip position of the controller, as defined by the SDK / Unity.
         /// </summary>
-        public Quaternion Rotation { get; set; }
+        T Grip { get;}
```

Comment:
Not all controllers will have grip.  Not a fan of the changes to this class. It's not as generic as it used to be.